storage-proofs-porep/src/stacked/vanilla/hash.rs

```diff
@@ -1,17 +1,22 @@
-use blstrs::Scalar as Fr;
-use filecoin_hashers::{POSEIDON_CONSTANTS_11, POSEIDON_CONSTANTS_2};
+use ff::PrimeField;
+use filecoin_hashers::{FieldArity, POSEIDON_CONSTANTS};
+use generic_array::typenum::{U11, U2};
 use neptune::poseidon::Poseidon;
 
 /// Hash all elements in the given column.
-pub fn hash_single_column(column: &[Fr]) -> Fr {
+pub fn hash_single_column<F: PrimeField>(column: &[F]) -> F {
```

Comment:
Note: make column hashing compatible with three fields; must lookup arity-11 Poseidon constants in a `typemap` table.